examples/b32_mul.rs

```diff
@@ -38,83 +38,74 @@ fn main() -> Result<()> {
+
+	let table_id = table.id();
+	let statement = Statement {
+		boundaries: vec![],
+		table_sizes: vec![test_vector.len()],
+	};
 
-	let in_a = binius_circuits::unconstrained::unconstrained::<BinaryField32b>(
-		&mut builder,
-		"in_a",
-		log_n_muls,
-	)
-	.unwrap();
-
-	let in_b = binius_circuits::unconstrained::unconstrained::<BinaryField32b>(
-		&mut builder,
-		"in_b",
-		log_n_muls,
-	)
-	.unwrap();
-	let out = builder.add_committed("out", log_n_muls, BinaryField32b::TOWER_LEVEL);
-
-	if let Some(witness) = builder.witness() {
-		let in_a_witness = witness
-			.get::<BinaryField32b>(in_a)?
-			.as_slice::<BinaryField32b>();
-		let in_b_witness = witness
-			.get::<BinaryField32b>(in_b)?
-			.as_slice::<BinaryField32b>();
-		let mut out_witness = witness.new_column::<BinaryField32b>(out);
-
-		let out_scalars = out_witness.as_mut_slice::<BinaryField32b>();
-
-		for (&a, &b, out) in izip!(in_a_witness, in_b_witness, out_scalars) {
-			*out = a * b;
+	let trace_gen_scope = tracing::info_span!("generating trace").entered();
+	let allocator = Bump::new();
+	let mut witness = WitnessIndex::<PackedType<OptimalUnderlier128b, B128>>::new(&cs, &allocator);
+	let table_witness = witness.init_table(table_id, test_vector.len()).unwrap();
+	let segment = table_witness.full_segment();
+
+	{
+		let mut in_a_vals = segment.get_mut_as::<B32, _, 1>(in_a).unwrap();
+		let mut in_b_vals = segment.get_mut_as::<B32, _, 1>(in_b).unwrap();
+		let mut out_vals = segment.get_mut_as::<B32, _, 1>(out).unwrap();
+
+		for (i, (a, b)) in test_vector.iter().enumerate() {
+			let a_field = B32::new(*a);
+			let b_field = B32::new(*b);
+			let result = a_field * b_field;
+
+			in_a_vals[i] = a_field;
+			in_b_vals[i] = b_field;
+			out_vals[i] = result;
 		}
 	}
 
-	builder.assert_zero(
-		"b32_mul",
-		[in_a, in_b, out],
-		binius_math::ArithCircuit::from(
-			binius_math::ArithExpr::<binius_field::BinaryField1b>::Var(0usize)
-				* binius_math::ArithExpr::<binius_field::BinaryField1b>::Var(1usize)
-				- binius_math::ArithExpr::<binius_field::BinaryField1b>::Var(2usize),
-		)
-		.convert_field(),
-	);
-
 	drop(trace_gen_scope);
 
-	let witness = builder
-		.take_witness()
-		.expect("builder created with witness");
-	let constraint_system = builder.build()?;
-
-	let backend = make_portable_backend();
+	let ccs = cs.compile(&statement).unwrap();
+	let witness = witness.into_multilinear_extension_index();
 
 	let proof =
 		constraint_system::prove::<
-			U,
+			OptimalUnderlier128b,
 			CanonicalTowerFamily,
 			Groestl256,
 			Groestl256ByteCompression,
 			HasherChallenger<Groestl256>,
 			_,
-		>(&constraint_system, args.log_inv_rate as usize, SECURITY_BITS, &[], witness, &backend)?;
+		>(
+			&ccs, args.log_inv_rate as usize, SECURITY_BITS, &[], witness, &make_portable_backend()
+		)?;
 
 	println!("Proof size: {}", ByteSize::b(proof.get_proof_size() as u64));
 
 	constraint_system::verify::<
-		U,
+		OptimalUnderlier128b,
```

Comment:
you should use OptimalUnderlier instead of OptimalUnderlier128b

Reply:
Thanks! I was going back and forth with it. I thought I had a problem on unit-test-amd when I got an error. I don't remember exactly but it was something about that `M512`, so I just used this one. 

I will revert it back here and in other places. Although I might need to do that as one pass as a follow up  because there are others (and some even mix). Is that alright with you?